.github/workflows/codeowners.yml

```diff
@@ -0,0 +1,19 @@
+name: Ping maintainers
+
+on:
+  pull_request:
+    types: [opened, synchronize, ready_for_review]
+  push:
```

Comment:
How does this action work with `push` events ?

Reply:
I don't know. I figured that I wanted to re-run the action on every push for testing the initial setup. I guess that it'll comment on the PR after every push, but we'll have to find out

Reply:
>  I guess that it'll comment on the PR after every push, but we'll have to find out

No. `push` is for stuff that happens _outside_ of PRs. Can't request a review of a commit pushed to master.

Reply:
Does this action request reviews or post comments?

`ofborg` used to post comments and it was changed to review requests, IIRC people though it was to spammy.

Reply:
Oh, I understand. In that case, `push` is indeed kinda useless.

The action posts comments instead of requesting reviews (by design, see their README). There's a lot of configuring to be done, but I wanted to start simple. (Configuring as in "forking modifying the action", since it has no YAML options.)

Reply:
Posting comments may be a dealbreaker TBH.

Review requests are preferred as they can't be resent, posting/updating comments will notify every time.

Reply:
I agree that we want review requests. Since the action does not have a lot of code, this shouldn't be too hard to add. But we first need to decide whether we want this at all, or if we prefer some self-written custom solution of some sorts.

Reply:
Having looked at it a bit more I don't think using `sourcegraph/codenotify` is a good idea.

> https://github.com/sourcegraph/codenotify#why-use-codenotify
> There can be a CODENOTIFY file in any directory.

IMHO allowing `CODENOTIFY` files at arbitrary places in the tree that can't be easily inspected without tooling is a negative considering the size of our tree and number of contributors.

Reply:
We can always restrict ourselves to using only the top-level entry first and then discuss that issue later on. IMHO the experiences with large files (all-packages.nix, release notes, …) have taught that they are rather ugly to work with and distributing content from a central file to the place it actually refers to is a good idea.

Reply:
> We can always restrict ourselves to using only the top-level entry first then discuss that issue later on.

How?

If it can't be restricted via the tooling to the top-level file it need be discussed first.

> IMHO the experiences with large files (all-packages.nix, release notes, …) have taught that they are rather ugly to work with and distributing content from a central file to the place it actually refers to is a good idea.

Currently maintainers could be set via `meta.maintainers` or `CODEOWNERS`. It is in a _predictable_ position, either at the top or the bottom of the tree. It can be easily inspected manually or for `meta.maintainers` it can be queried it using `nix`.

With this approach it can be set anywhere, can't be easily inspected manually and it can only be queried via a custom tool?